### PR TITLE
Translation Memory fix

### PR DESCRIPTION
**Pull request by mathjazz**
Store Translation Memory on batch editing actions (approve and search & replace).

@jotes r?

Note: This is branched off #601.

### CODE REVIEW

File: pontoon/base/models.py [hunk 1]
```diff
@@ -2025,29 +2025,30 @@ def save(self, imported=False, *args, **kwargs):
...
-        attribute and, if it does, see if this translation is more
-        recent than it. If so, replace it and save.
+        Set `latest_translation` to this translation if its more recent than
+        the currently stored translation. Do this for all affected models.
         """
-        latest = instance.latest_translation
-        if latest is None or self.latest_activity['date'] > latest.latest_activity['date']:
-            instance.latest_translation = self
-            instance.save(update_fields=['latest_translation'])
+        resource = self.entity.resource
```
Comment (jotes) [inline pontoon/base/models.py]: I wonder if we just use `.update()` e.g.
```
Project.objects.filter(pk=resource.project_id, latest_activity__date__lte=self.latest_activity['date'])
Locale.objects.filter(pk=self.locale_id ....
```

File: pontoon/base/views.py [hunk 2]
```diff
@@ -309,7 +311,10 @@ def get_translations_info(translations):
         replace = request.POST.get('replace')
 
         try:
-            translations = translations.find_and_replace(find, replace, request.user)
+            translations, changed = translations.find_and_replace(find, replace, request.user)
+            changed_translation_pks = [c.pk for c in changed]
+            if changed:
+                latest_translation_pk = changed[-1].pk
```
Comment (jotes) [inline pontoon/base/views.py]: `max(changed)`?

File: pontoon/base/models.py (hunk 1 above)
Comment (mathjazz) [inline pontoon/base/models.py]: Just curious - did you (as part of this PR) also review the entire #601, or do you still want to do that separately?

**Comment by mathjazz** [comment]
Updated. I'm not exactly sure if you reviews both PRs here (#603 *and* #601), or do you still want to review the later separately.

**Comment by jotes** [comment]
For some reason, I started to look at them in reverse order.

**Pull request merged**